Release Zebra 1.5.0 (#8027)
ZcashFoundation/zebra: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org).

## [Zebra 1.5.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.5.0) - 2023-11-28

This release avoids a panic when reading cached sprout or history trees in Zebra v1.4.0, 
improves how Zebra recovers from network interruptions, avoids potential network hangs, and

> limits the ability of syntethic nodes to spread throughout the network through Zebra to

Author: ```suggestion
limits the ability of synthetic nodes to spread throughout the network through Zebra to
```

> address some of the Ziggurat's report.

Author: ```suggestion
address some of the Ziggurat red team report.
```

Finally, we've added an experimental "shielded-scan" feature and the zebra-scan crate as steps

> towards supporting shielded scanning in Zebra.

Reviewer: There are multiple known security risks/issues in this code, do we need to warn people about not using important private keys with it?

There is a security list in #7728 but it's probably incomplete.

### Security

- security(net): Stop sending peer addresses from version messages directly to the address book ([#7977](https://github.com/ZcashFoundation/zebra/pull/7977))
- security(net): Limit how many addresses are sent directly to the address book for a single peer address message ([#7952](https://github.com/ZcashFoundation/zebra/pull/7952))
- security(net): Rate-limit GetAddr responses to avoid sharing the entire address book over a short period ([#7955](https://github.com/ZcashFoundation/zebra/pull/7955))

### Added

- feat(config): Add config field for the viewing keys used by zebra-scan ([#7949](https://github.com/ZcashFoundation/zebra/pull/7949))
- feat(scanner): Add basic RAM database to store keys and scan results ([#7942](https://github.com/ZcashFoundation/zebra/pull/7942))
- feat(scanner): Spawn zebra-scan task from zebrad with configured viewing keys ([#7989](https://github.com/ZcashFoundation/zebra/pull/7989))

> 

Author: ```suggestion
- poc(scanner): Proof of concept for shielded scanning ([#7758](https://github.com/ZcashFoundation/zebra/pull/7758))

```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org).

## [Zebra 1.5.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.5.0) - 2023-11-28

> This release avoids a panic when reading cached sprout or history trees in Zebra v1.4.0, 

Reviewer: It's important to tell users that this panic is rare, and only happened from 1.4.0.

```suggestion
This release fixes a panic that was introduced in Zebra v1.4.0, which happens in rare circumstances when reading cached sprout or history trees. 
```

improves how Zebra recovers from network interruptions, avoids potential network hangs, and
limits the ability of synthetic nodes to spread throughout the network through Zebra to

> address some of the Ziggurat red team report.

Reviewer: Readability note: this sentence has approximately 50 words. Sentences of around 10-15 words are more readable.

In particular, some parts of the sentence relate to the Ziggurat report, and some don't.

Finally, we've added an experimental "shielded-scan" feature and the zebra-scan crate as steps
towards supporting shielded scanning in Zebra. There are still known security issues with this
feature, it is not recommended to use important private keys when testing Zebra's experimental
shielded scanning support. Ongoing development is tracked in issue [#7728](https://github.com/ZcashFoundation/zebra/issues/7728).

> As of this release, Zebra requires Rust 1.73 to build.

Reviewer: Security advisories should be short, explicit, and clearly identified. As well as telling people what not to do, it helps to provide safer alternatives.

It is possible for someone to skim this paragraph and miss the security advice. They could also think that it's just recommended advice, or that their keys aren't important.

```suggestion
As of this release, Zebra requires Rust 1.73 to build.

Finally, we've added an experimental "shielded-scan" feature and the zebra-scan crate as steps
towards supporting shielded scanning in Zebra. This feature has known security issues.
It is for experimental use only. Ongoing development is tracked in issue [#7728](https://github.com/ZcashFoundation/zebra/issues/7728).

### Important Security Warning

Do not use regular or sensitive viewing keys with Zebra's experimental scanning feature. Do not use this
feature on a shared machine. We suggest generating new keys for experimental use.
```

### Security

- security(net): Stop sending peer addresses from version messages directly to the address book ([#7977](https://github.com/ZcashFoundation/zebra/pull/7977))
- security(net): Limit how many addresses are sent directly to the address book for a single peer address message ([#7952](https://github.com/ZcashFoundation/zebra/pull/7952))
- security(net): Rate-limit GetAddr responses to avoid sharing the entire address book over a short period ([#7955](https://github.com/ZcashFoundation/zebra/pull/7955))

### Added

- feat(config): Add config field for the viewing keys used by zebra-scan ([#7949](https://github.com/ZcashFoundation/zebra/pull/7949))

> - feat(scan): Add basic RAM database to store keys and scan results ([#7942](https://github.com/ZcashFoundation/zebra/pull/7942))

Reviewer: Optional: The RAM database code was added and deleted before being released, so let's combine this PR with the replacement database PR, or clarify the description?

Author: Yep, applied in 
https://github.com/ZcashFoundation/zebra/pull/8027/commits/efd8085da831f3f7c80325827beddc77c379765e.

- change(scan): Use the on-disk database for keys and results ([#8036](https://github.com/ZcashFoundation/zebra/pull/8036))
- change(scan): Store scanned TXIDs in "display order" ([#8057](https://github.com/ZcashFoundation/zebra/pull/8057))
- change(scan): Create a function that scans one block by height, and stores the results in the database ([#8045](https://github.com/ZcashFoundation/zebra/pull/8045))
- change(scan): Store one transaction ID per database row, to make queries easier ([#8062](https://github.com/ZcashFoundation/zebra/pull/8062)

### Fixed

- fix(db): Fix a sprout/history tree read panic in Zebra v1.4.0, which only happens before the 25.3.0 state upgrade completes ([#7972](https://github.com/ZcashFoundation/zebra/pull/7972))
- fix(net): Fix potential network hangs, and reduce code complexity ([#7859](https://github.com/ZcashFoundation/zebra/pull/7859))
- fix(scan): Start scanning task only if there are keys to scan ([#8059](https://github.com/ZcashFoundation/zebra/pull/8059))

### Contributors

Thank you to everyone who contributed to this release, we couldn't make Zebra without you:

> @arya2, @dependabot[bot], @oxarbitrage, @teor2345 and @upbqdn

Reviewer: Optional: The bot does not need thanks 🙂 

```suggestion
@arya2, @oxarbitrage, @teor2345 and @upbqdn
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org).

## [Zebra 1.5.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.5.0) - 2023-11-28

This release fixes a panic that was introduced in Zebra v1.4.0, which happens in rare circumstances when reading cached sprout or history trees. 
improves how Zebra recovers from network interruptions, avoids potential network hangs, and
limits the ability of synthetic nodes to spread throughout the network through Zebra to

> address some of the Ziggurat red team report.

Author: ```suggestion
This release:
- fixes a panic that was introduced in Zebra v1.4.0, which happens in rare circumstances when reading cached sprout or history trees. 
- further improves how Zebra recovers from network interruptions and prevents potential network hangs.
- limits the ability of synthetic nodes to spread throughout the network through Zebra to address some of the Ziggurat red team report.
```

Reviewer: Looks good, thanks!
